testing/integration-tests/proc-macro/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+extern crate proc_macro;
+use proc_macro::TokenStream;
+
+use quote::{format_ident, quote};
+use syn::{
+    parse::{Parse, ParseStream},
+    Error,
+};
+
+#[proc_macro_attribute]
+pub fn subxt_test(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let subxt_attr = match syn::parse::<SubxtTestAttr>(attr) {
+        Ok(subxt_attr) => subxt_attr,
+        Err(err) => return err.into_compile_error().into(),
+    };
+    let timeout_duration = subxt_attr.timeout.unwrap_or(60 * 5);
```

Comment:
I wonder how many tests actually require longer than this? Since its our macro, we might as well set default in it that we are happy using for all of our tests. I can see many (all?) tests have 800s timeouts set!